app/views/contexts/summary/list_collector_block.py

```diff
@@ -0,0 +1,235 @@
+            list_store=list_store,
+            metadata=metadata,
+            response_metadata=response_metadata,
+            schema=schema,
+        )
+        self._list_store = list_store
+        self._schema = schema
+        self._location = location
+        self._section: dict = self._schema.get_section(self._location.section_id) or {}
+        self._language = language
+        self._answer_store = answer_store
+        self._metadata = metadata
+        self._response_metadata = response_metadata
+        self._routing_path = routing_path
+        self._progress_store = progress_store
+
+    # pylint: disable=too-many-locals
+    def list_summary_element(self, summary: Mapping[str, Any]) -> dict[str, Any]:
+        list_collector_block = None
+        (
+            edit_block_id,
+            remove_block_id,
+            primary_person_edit_block_id,
+            related_answers,
+            item_label,
+            item_anchor,
+        ) = (None, None, None, None, None, None)
+        current_list = self._list_store[summary["for_list"]]
+
+        list_collector_blocks = list(
+            self._schema.get_list_collectors_for_list(
+                self._section, for_list=summary["for_list"]
+            )
+        )
+
+        add_link = self._add_link(summary, list_collector_block)
+
+        list_collector_blocks_on_path = [
+            list_collector_block
+            for list_collector_block in list_collector_blocks
+            if list_collector_block["id"] in self._routing_path.block_ids
+        ]
+
+        list_collector_block = (
+            list_collector_blocks_on_path[0]
+            if list_collector_blocks_on_path
+            else list_collector_blocks[0]
+        )
+
+        rendered_summary = self._placeholder_renderer.render(
+            summary, self._location.list_item_id
+        )
+
+        section_id = self._section["id"]
+        if list_collector_blocks_on_path:
+            edit_block_id = list_collector_block["edit_block"]["id"]
+            remove_block_id = list_collector_block["remove_block"]["id"]
+            add_link = self._add_link(summary, list_collector_block)
+            related_answers = self._get_related_answers(current_list)
+            item_anchor = self._schema.get_item_anchor_answer_id(
+                section_id, current_list.name
+            )
+            item_label = self._schema.get_item_label(section_id, current_list.name)
+
+        if len(current_list) == 1 and current_list.primary_person:
+            if primary_person_block := self._schema.get_list_collector_for_list(
+                self._section, for_list=summary["for_list"], primary=True
+            ):
+                primary_person_edit_block_id = edit_block_id = primary_person_block[
+                    "add_or_edit_block"
+                ]["id"]
+        list_summary_context = self.list_context(
+            list_collector_block["summary"],
+            for_list=list_collector_block["for_list"],
```

Comment:
A new line would help readability. It's crammed.

Reply:
Added 👍